extension/src/frequency.rs

```diff
@@ -430,6 +536,56 @@ pub fn topn_agg_with_skew_trans(
         {SpaceSavingTransState::topn_agg_from_type_id(skew, n as u32, typ, collation)}).internal()
 }
 
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+pub fn topn_agg_with_skew_bigint_trans(
+    state: Internal,
+    n: i32,
+    skew: f64,
+    value: Option<i64>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal> {
+    if n <= 0 {
```

Comment:
We can avoid repeating these checks by moving them into topn_agg_from_type_id as show in commit c9eb12fe91eb232ac557500d0c0513921e1f351c